nix/cells/automation/pipelines.nix

```diff
@@ -0,0 +1,103 @@
+{ cell
+, inputs
+}:
+let
+  inherit (inputs.nixpkgs) system;
+
+  common =
+    { config
+    , ...
+    }: {
+      preset = {
+        nix.enable = true;
+
+        github-ci = {
+          # Tullia tasks can run locally or on Cicero.
+          # When no facts are present we know that we are running locally and vice versa.
+          # When running locally, the current directory is already bind-mounted
+          # into the container, so we don't need to fetch the source from GitHub
+          # and we don't want to report a GitHub status.
+          enable = config.actionRun.facts != { };
+          repo = "input-output-hk/plutus";
+        };
+      };
+    };
+in
+{
+  ci = { config, lib, ... }: {
+    imports = [ common ];
+
+    preset.github-ci.sha = config.preset.github-ci.lib.readRevision
+      inputs.cells.cloud.library.actions.ci.input
+      null;
+
+    command.text = ''
+      nix build -L .#${lib.escapeShellArg system}.automation.ciJobs.required
+    '';
+
+    memory = 1024 * 8;
+    nomad.resources.cpu = 10000;
+  };
+
+  publish-documents = { config, pkgs, lib, ... }: {
+    imports = [ common ];
+
+    preset.github-ci.sha = config.preset.github-ci.lib.readRevision
+      inputs.cells.cloud.library.actions.documents.input
+      null;
+
+    command.text =
+      let
+        documents = __mapAttrs (_: v: "${v}.pdf") {
+          plutus-report = "plutus-core-specification";
+          plutus-core-spec = "plutus-core-specification";
+          extended-utxo-spec = "extended-utxo-specification";
+          unraveling-recursion-paper = "unraveling-recursion";
+          system-f-in-agda-paper = "paper";
+          eutxo-paper = "eutxo";
+          utxoma-paper = "utxoma";
+          eutxoma-paper = "eutxoma";
+        };
+      in
+      ''
+        set -x
+
+        nix build -L ${lib.escapeShellArgs (map (k: ".#${k}") (__attrNames documents))}
+
+        revision=$(${lib.escapeShellArg config.preset.github-ci.sha})
+      '' + __concatStringsSep "\n" (lib.imap0
+        (i: k: ''
+          jq > ${lib.escapeShellArg k}.json --null-input \
+            --arg revision "$revision" \
+            --arg document ${lib.escapeShellArg k} \
+            '{"plutus/document/\($document)": $revision}'
+
+          http --multipart --ignore-stdin "$CICERO_WEB_URL/api/run/$NOMAD_JOB_ID/fact" \
+            value@${lib.escapeShellArg k}.json \
+            binary@result${
+              lib.optionalString (i != 0) "-${toString i}"
+            }/${lib.escapeShellArg documents.${k}}
```

Comment:
Took me quite a lot of staring to work out what's going on here. You're getting the `result` symlinks and using them as the things to provide as the result, I think? Are we guaranteed that they come out in the right order? That's quite non-obvious to me, and I don't see it guaranteed anywhere in the docs.

I wonder if it would be simpler to just do the whole operation once per document, including calling `nix build`. Then we can use `-o` to get a specific known output link.

Also something it took me a while to realise and I think is maybe worth documenting: I was wondering why we can't just splice in a reference to the document here and avoid doing this convoluted dance. But then I realised that that would make us build the doc when we build this task definition, which is _not_ when we want it to happen. So we have to do it like this.

Reply:
That said, this has lingered on a bit and we can refactor this afterwards I guess.

Reply:
> Are we guaranteed that they come out in the right order?

AFAIK yes.

`nix build .#{a,b,c}` will result in

```
result → /nix/store/…-a
result-1 → /nix/store/…-b
result-2 → /nix/store/…-c
```

This is documented in `nix build --help` in the "Description" section.